### PR TITLE
[FLINK-18598][python][docs] Add instructions for asynchronous execute in PyFlink doc.

### PR DESCRIPTION
**Pull request by shuiqiangchen**
## What is the purpose of the change

Add instructions for asynchronous execute in PyFlink doc

## Brief change log

- Added brief instructions for asynchronous execute in PyFlink FAQ page (faq.md).

## Verifying this change

This pull request is a documentation enhancement without test case coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: docs/dev/python/faq.md [hunk 1]
```diff
@@ -52,6 +52,19 @@ $ source venv/bin/activate
 $ python xxx.py
 {% endhighlight %}
 
+**Note:** When performing asynchronous instructions in a LocalStreamEnvironment, remember to block the process, or it
```
Comment (HuangXingBo) [inline docs/dev/python/faq.md:55]:
```suggestion
**Note:** If you run jobs in mini-cluster asynchronously, you need to make use of `JobClient` to wait for jobs to finish.
```

**Review by HuangXingBo — COMMENTED** [review]
@shuiqiangchen Thanks a lot for the PR. I only left a minor comment.

**Comment by shuiqiangchen** [comment]
@HuangXingBo Thank you for your comments, I have updated the PR according to your comments, please have a look.

**Comment by HuangXingBo** [comment]
@shuiqiangchen Thanks for the PR. LGTM

File: docs/dev/python/faq.md [hunk 2]
```diff
@@ -52,6 +52,20 @@ $ source venv/bin/activate
 $ python xxx.py
 {% endhighlight %}
 
+**Note:** When running jobs in mini-cluster asynchronously, remember to  get a `JobClient` to wait for the execution to
```
Comment (dianfu) [inline docs/dev/python/faq.md:55]:
```suggestion
**Note:** When executing jobs in mini-cluster asynchronously (e.g. TableEnvironment.execute_sql, StatementSet.execute, etc in the Python Table API; StreamExecutionEnvironment.execute_async in the Python DataStream API), remember to wait for the execution to 
```

**Comment by shuiqiangchen** [comment]
@dianfu Highly appreciate for your suggestions, I have updated the PR, please have a look.

**Review by dianfu — APPROVED** [review]

**Pull request closed without merging**